great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -93,6 +94,33 @@ class SQLDatasourceError(Exception):
     pass
 
 
+class SQLAlchemyCreateEngineError(SQLDatasourceError):
+    """
+    An error creating a SQLAlchemy `Engine` object.
+
+    Not to be confused with the GX `SQLAlchemyExecutionEngine`.
+    """
+
+    @overload
+    def __init__(self, addendum: str | None = ..., cause: Exception = ...): ...
+
+    @overload
+    def __init__(self, addendum: str = ..., cause: Exception | None = ...): ...
+
+    def __init__(
+        self,
+        addendum: str | None = None,
+        cause: Exception | None = None,
+    ):
+        """Must provide a `cause`, `addendum`, or both."""
+        message = "Unable to create SQLAlchemy Engine"
+        if cause:
+            message += f": due to {cause!r}"
+        if addendum:
+            message += f": {addendum}"
+        super().__init__(message)
```

Comment:
@anthonyburdi This is how I'm addressing the `TRY003` warnings.

Reply:
@Kilo59 thanks for the ping. One question: Should there be validation based on the docstring? 
```
"""Must provide a `cause`, `addendum`, or both."""
```
I'm thinking no, because we wouldn't want the exception to raise an exception.

Reply:
It's enforced via the type-checker and the declared `overloads`.